src/OpenTelemetry.Exporter.Geneva/Metrics/GenevaMetricExporter.cs

```diff
@@ -268,7 +268,11 @@ protected override void Dispose(bool disposing)
         {
             try
             {
-                this.metricDataTransport?.Dispose();
+                // MetricEtwDataTransport is a singleton and should not be disposed
+                if (this.metricDataTransport is not MetricEtwDataTransport)
+                {
+                    this.metricDataTransport?.Dispose();
```

Comment:
My plan was to test with EventListener, however, I got some unexpected data (error message instead of the binary data sent into ETW) in the payload.

Sample code:
```
private class TestMetricEtwDataListener : EventListener
{
    private readonly Action<EventWrittenEventArgs> eventHandler;

    public TestMetricEtwDataListener(Action<EventWrittenEventArgs> eventHandler)
    {
        this.eventHandler = eventHandler;
    }

    protected override void OnEventSourceCreated(EventSource eventSource)
    {
        if (eventSource.Name == "OpenTelemetryGenevaMetricExporter")
        {
            this.EnableEvents(eventSource, EventLevel.Verbose);
        }

        base.OnEventSourceCreated(eventSource);
    }

    protected override void OnEventWritten(EventWrittenEventArgs eventData)
    {
        if (eventData.EventSource.Name != "OpenTelemetryGenevaMetricExporter")
        {
            return;
        }

        for (int i = 0; i < eventData.Payload.Count; ++i)
        {
            object obj = eventData.Payload[i];
            this.testOutputHelper.WriteLine($"Type is: `{obj.GetType()}`.");
            this.testOutputHelper.WriteLine($"EventData.Payload[{i}] is: `{obj}`.");

            Debug.WriteLine($"EventData.Payload[{i}] is {obj}");
        }
    }
}
```

The output is:
```
Type is: `System.String`.
EventData.Payload[0] is: `Event 70 was called with 1 argument(s), but it is defined with 0 parameter(s).`.
Type is: `System.String`.
EventData.Payload[0] is: `Event 70 was called with 1 argument(s), but it is defined with 0 parameter(s).`.
```

Looks like it's due to the way we define the `[Event]` attribute. (I found the definition of the error message and where the error is thrown. The definition of the error message is [here](https://github.com/microsoft/referencesource/blob/51cf7850defa8a17d815b4700b67116e3fa283c2/mscorlib/mscorlib.txt#L1023C1-L1023C37). And it's used [here](https://source.dot.net/#System.Private.CoreLib/artifacts/obj/coreclr/System.Private.CoreLib/linux.x64.Debug/System.SR.cs,98cb18dbb6afb4c7,references).)

The metrics exporter use TLV based serialization to send data to ETW. However, EventSource would send error message to EventListener's callback.

Reply:
I'll look for other ways to directly read the binary bytes of the ETW event without EventListener.

Reply:
Considering this involves the external components (ETW mechanisms), it is more of an integration test to validate with an ETW consumer tool like `EventListener`.

I've added a unit test which focuses on that the `MetricEtwDataTransport` singleton instance is shared and NOT disposed among multiple instances of `GenevaMetricExporter`.